clang/lib/Sema/SemaTemplate.cpp

```diff
@@ -8343,58 +8343,52 @@ bool Sema::CheckTemplateTemplateArgument(TemplateTemplateParmDecl *Param,
   // C++1z [temp.arg.template]p3: (DR 150)
   //   A template-argument matches a template template-parameter P when P
   //   is at least as specialized as the template-argument A.
-  // FIXME: We should enable RelaxedTemplateTemplateArgs by default as it is a
-  //  defect report resolution from C++17 and shouldn't be introduced by
-  //  concepts.
-  if (getLangOpts().RelaxedTemplateTemplateArgs) {
```

Comment:
This is just being deprecated, right?  Don't we still need this?

Reply:
Only the flag is deprecated, we want the new 'on' behavior to stay.

Reply:
It's a removal of the flag's behavior, with the addition of provisional wording changes, in order to moot the original reason the flag existed.

I see where the confusion comes from. The original patch did not remove the flag completely, just changed the default, but still deprecated it. After some offline discussion with @zygoloid, we decided to remove the flag as well, but I forgot to update the title.

@zygoloid thoughts on that?

How about for starters we just change the default, but still deprecate the flag both ways anyway?
If the transition is a concern, I think it would be a problem that we don't start to warn the users of the flags as soon as possible. Otherwise, I think this will be specially problematic if there are current users of the negative spelling `-fno-relaxed-template-template-args`.

Reply:
>How about for starters we just change the default, but still deprecate the flag both ways anyway?
If the transition is a concern, I think it would be a problem that we don't start to warn the users of the flags as soon as possible. Otherwise, I think this will be specially problematic if there are current users of the negative spelling -fno-relaxed-template-template-args.

Seems reasonable to me.  

I had missed the impact of the CWG issue, if @zygoloid thinks we're Ok with just deprecating the flags (or removing them?  I think I'd prefer the deprecate-but-do-nothing over removal), I'm Ok too.

Reply:
I think the goal should be to remove the flags eventually, but given that it's easy to keep the flags functioning, I think it'd be a good idea to do so for at least long enough for people to tell us if this change breaks something. Maybe one release cycle with the default flipped and the flags working but warning, then we remove them if people aren't complaining?

Reply:
I also wonder if, at least for the time being, we should leave C++03 alone (despite the paper being a DR)

Reply:
> I think the goal should be to remove the flags eventually, but given that it's easy to keep the flags functioning, I think it'd be a good idea to do so for at least long enough for people to tell us if this change breaks something. Maybe one release cycle with the default flipped and the flags working but warning, then we remove them if people aren't complaining?

Thats was exactly my thought, thanks!  I would like that too.


>I also wonder if, at least for the time being, we should leave C++03 alone (despite the paper being a DR)

I think we should do it whole-hog or not at all.  I don't see how 03 is significantly different here, and it would likely result in us delaying removing the flags an additional release cycle.

Reply:
Yes, I think if there are problems in existing C++03 code, we should know about it, I can't see how that same problem wouldn't necessarily apply to newer standards.